### PR TITLE
Extend the tooltip decorator

### PR DESCRIPTION
**Pull request by philearley**
# Description

* Allows users to interact with the tooltip. We do not hide the tooltip if the person driving the browser moves their cursor over into the tooltip, it will ensure the tooltip remains in view.

# Screenshots / Gifs

tooltip-decorator

### CODE REVIEW

File: src/utils/decorators/tooltip-decorator/tooltip-decorator.js [hunk 1]
```diff
@@ -196,8 +198,12 @@ const TooltipDecorator = (ComposedComponent) => {
      * @return {void}
      */
     onHide = () => {
-      clearTimeout(this._tooltipTimeout);
-      this.setState({ isVisible: false });
+      clearTimeout(this._showTooltipTimeout);
+
+      this._hideTooltipTimeout = setTimeout(() => {
+        this.setState({ isVisible: false });
+        this.positionTooltip();
```
Comment (andrewjtait) [inline src/utils/decorators/tooltip-decorator/tooltip-decorator.js]: I don't think you need to position the tooltip on hide

File: src/utils/decorators/tooltip-decorator/tooltip-decorator.js [hunk 2]
```diff
@@ -183,7 +183,9 @@ const TooltipDecorator = (ComposedComponent) => {
      * @return {void}
      */
     onShow = () => {
-      this._tooltipTimeout = setTimeout(() => {
+      clearTimeout(this._hideTooltipTimeout);
+
+      this._showTooltipTimeout = setTimeout(() => {
```
Comment (andrewjtait) [inline src/utils/decorators/tooltip-decorator/tooltip-decorator.js]: as this property has been renamed, you should update or remove line 119

**Comment by philearley** [comment]
Force pushed due to the original PR being off of master.

**Comment by edleeks87** [comment]
will QA in staging environment with the tag

**Pull request merged**